pkg/module/sync/runner.go

```diff
@@ -142,25 +146,46 @@ func (r *Runner) patchManifest(ctx context.Context, owner string, manifestObj *v
 	return nil
 }
 
-func (r *Runner) updateAvailableManifestSpec(ctx context.Context, manifestObj *v1beta2.Manifest) error {
+func (r *Runner) updateAvailableManifestSpec(ctx context.Context, manifestObj *v1beta2.Manifest,
+) error {
 	manifestInCluster := &v1beta2.Manifest{}
 
 	if err := r.Get(ctx, client.ObjectKey{Namespace: manifestObj.GetNamespace(), Name: manifestObj.GetName()},
 		manifestInCluster); err != nil {
 		return fmt.Errorf("error get manifest %s: %w", client.ObjectKeyFromObject(manifestObj), err)
 	}
-	if !needToUpdate(manifestInCluster, manifestObj) {
-		return nil
-	}
 	manifestInCluster.Spec = manifestObj.Spec
 	if err := r.Update(ctx, manifestInCluster); err != nil {
 		return fmt.Errorf("error update manifest %s: %w", client.ObjectKeyFromObject(manifestObj), err)
 	}
 	return nil
 }
 
-func needToUpdate(manifestInCluster, manifestObj *v1beta2.Manifest) bool {
-	return manifestInCluster.Spec.Version != manifestObj.Spec.Version
+func NeedToUpdate(kymaStatus v1beta2.KymaStatus, manifestObj *v1beta2.Manifest) bool {
+	moduleStatus := getModuleStatus(kymaStatus, manifestObj.GetName())
+
+	if moduleStatus == nil {
+		return true
+	}
+
+	return moduleStatus.Version != manifestObj.Spec.Version ||
+		moduleStatus.Channel != getChannelLabel(manifestObj)
```

Comment:
Can you create an issue to introduce channel as an optional field in manifest CR v1beta2, we need to get rid of this get channel from label logic asap.

Reply:
https://github.com/orgs/kyma-project/projects/30/views/13?pane=issue&itemId=60056896